### PR TITLE
[5143] ArticlePage Frontend

### LINKED ISSUE #5143: Publication Article Page frontend
it exists, make it pretty. there is a lot to check here in terms of the various components as designed and where they should display on a page.

### PR DESCRIPTION
**Pull request by KalobTaulien**
Closes #5143 
Related PRs/issues #5143 

## Checklist

**Tests**
- [ ] Is the code I'm adding covered by tests?
- [ ] Browser tested 
- [ ] Has desktop, tablet and mobile responsiveness

### CODE REVIEW

**Review by Zenix27 — APPROVED** [review]

File: network-api/networkapi/wagtailpages/pagemodels/article_fields.py
Comment (easherma) [inline network-api/networkapi/wagtailpages/pagemodels/article_fields.py:1]: good idea to split this out!

File: network-api/networkapi/wagtailpages/templates/wagtailpages/blocks/article_table_block.html
Comment (easherma) [inline network-api/networkapi/wagtailpages/templates/wagtailpages/blocks/article_table_block.html:7]: great comment! lots of good template comments

File: source/images/right-chevron.svg
Comment (easherma) [inline source/images/right-chevron.svg:1]: I'd want to hear from @cadecairos or @Pomax on this, should these live here, or maybe under images/glyps? Or are there any icon dependencies in the project that might already have chevron arrows for the links?

File: source/sass/components/article-page.scss
Comment (easherma) [inline source/sass/components/article-page.scss:1]: There is potentially alot of technical debt in this css. Some of the values being pulled, like font-family, are already defined as variables. 

The same may be true for some of these colors, and for the ones where they aren't, ideally we'd check with design and verify that they are adding new colors to the palette for the site, make new variables, and add them to the style guide.

As much as possible, we should use the Mozilla Bootstrap theme to keep things consistent. It's totally possible that there are aspects of the design spec that don't match what is defined for the site at large; for better long term maintenance it would be best for the Mozilla team to be able to reconcile all these discrepancies. 

All that being said, given our time frame it may be better to not count these issues as blocking.

**Review by easherma — COMMENTED** [review]
see comments. wasn't going to approve since this is still marked as a draft

File: network-api/networkapi/templates/pages/base.html
Comment (easherma) [inline network-api/networkapi/templates/pages/base.html:34]: if this is something new needed for design, maybe they want a change to say one of the other font weights is fine

File: network-api/networkapi/wagtailpages/pagemodels/publications/article.py
Comment (easherma) [inline network-api/networkapi/wagtailpages/pagemodels/publications/article.py:76]: For QA, want to make sure this behavior matches: https://github.com/mozilla/foundation.mozilla.org/issues/5145#issuecomment-683946332

File: network-api/networkapi/wagtailpages/pagemodels/publications/article.py
Comment (easherma) [inline network-api/networkapi/wagtailpages/pagemodels/publications/article.py:109]: i'd be interested to see what affect getting these titles this way has on performance *nerd glasses emoji*

File: source/images/right-chevron.svg
Comment (cadecairos) [reply source/images/right-chevron.svg:1]: looks like there's already a lot of disorganization in `source/images/`. However, I do see a `down-chevron.svg` in `glyphs` so we should probably move this there.

File: source/sass/components/article-page.scss
Comment (cadecairos) [reply source/sass/components/article-page.scss:1]: I'd defer to @mmmavis on situations like this. If we need to get this merged to unblock other PRs, let's make sure we open a follow-up issue to discuss potential improvements and/or work to stick to the styleguide

**Comment by easherma** [comment]
@KalobTaulien when running locally, I got this error trying to view a fake article page. Maybe cause it is missing some data? Should the template be able to handle this sort of exception?
err

File: network-api/networkapi/wagtailpages/pagemodels/publications/article.py
Comment (KalobTaulien) [reply network-api/networkapi/wagtailpages/pagemodels/publications/article.py:109]: Should be very little. All stream data is kept in a single field as a dictionary, then we loop through it to find "content" streamblocks, and use BeautifulSoup to pull out the `h2` elements. There are no additional queries, the biggest thing is BS4 parsing the HTML (Or rather, the "wagtail" HTML) and that should be quite minimal as it's not ever saved as deeply nested HTML. That said, I'd be interested to see the performance on this as well. That could be a spike ticket though.

**Comment by KalobTaulien** [comment]
> @KalobTaulien when running locally, I got this error trying to view a fake article page. Maybe cause it is missing some data? Should the template be able to handle this sort of exception?

@easherma I can't seem to replicate that. Does this happen still when you create a new env with `inv new-env`?

File: network-api/networkapi/templates/pages/base.html
Comment (KalobTaulien) [reply network-api/networkapi/templates/pages/base.html:34]: Good question. I asked the design team in Slack.

File: network-api/networkapi/wagtailpages/pagemodels/publications/article.py
Comment (KalobTaulien) [reply network-api/networkapi/wagtailpages/pagemodels/publications/article.py:76]: I've re-written the functions based on the #5145 comment. I think they're small enough now and different enough that if we wanted to write tests for them it'd be easier as two functions.

File: network-api/networkapi/templates/pages/base.html
Comment (KalobTaulien) [reply network-api/networkapi/templates/pages/base.html]: We're not using 500. Ill have that removed shortly.

File: source/js/dropdowns.js
Comment (Pomax) [inline source/js/dropdowns.js:14]: Handy bit of API: `querySelectorAll` returns an array-like that does not offer `map`, but _does_ offer `forEach`:

```suggestion
    document.querySelectorAll(`.dropdown-toggle`).forEach(element =>
      element.addEventListener(`click`,  _ => {
        element.classList.toggle("show");
        const target = document.querySelector(`[aria-labelledby="${element.id}"]`);
        target.classList.toggle("d-block");
      })
    );
```

Also note that if we use proper `<button>` markup, there is no default behaviour that needs preventing.

File: source/js/dropdowns.js
Comment (Pomax) [inline source/js/dropdowns.js:21]: Same kind of rewrite as above.

**Review by Pomax — CHANGES_REQUESTED** [review]

**Comment by Pomax** [comment]
It might be a good idea to squash-rebase before working on this some more, because 35 commits is definitely too many commits to easily diff between =)

**Comment by Pomax** [comment]
I've also told heroku to rebuild the review app, so there should be a https://foundation-s-5143-artic-6u4msf.herokuapp.com up for the next few days.

File: network-api/networkapi/wagtailpages/factory/publication.py
Comment (KalobTaulien) [reply network-api/networkapi/wagtailpages/factory/publication.py:104]: I think this seeding is good where it is. It does the job it's supposed to do. Ill update the docstring.

File: network-api/networkapi/wagtailpages/pagemodels/publications/article.py
Comment (KalobTaulien) [reply network-api/networkapi/wagtailpages/pagemodels/publications/article.py:76]: Roger that.

File: network-api/networkapi/utility/templatetags/remove_richtext_wrapping.py
Comment (KalobTaulien) [reply network-api/networkapi/utility/templatetags/remove_richtext_wrapping.py:9]: To remove styling from other rich-text areas on the site. This is used in the image captions on the Article Page, which is a richtext area, but we dont want all the default .rich-text styling applied to it. However, we want to keep links, bold, and italics.

**Pull request merged**
